numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1330,6 +1339,8 @@ _new_sortlike(PyArrayObject *op, int axis, PyArray_SortFunc *sort,
 
         PyArray_ITER_NEXT(it);
     }
+    
+    NPY_AUXDATA_FREE(auxdata);
```

Comment:
I think I'd do the free in `PyArray_Sort` - it is weird if a function "eats" your `auxdata`. Has the additional benefit that you have to do it only once.

Reply:
Yes, that makes sense! I've moved things to `PyArray_Sort`.